TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -951,6 +953,203 @@ private void FinalCompilationStepCompleted(object cUnit, ProgramClassEvent progr
+
+                // Name (preceded by an indent depending on the level)
+                AppendToRow(GetIndent(dataDefinition) + GetName(dataDefinition));
+
+                // Declaration (Picture, Usage, REDEFINES, OCCURS, ...)
+                AppendToRow(GetDeclaration(dataDefinition));
+
+                // Start/End/Length
+                var start = dataDefinition.StartPosition;
+                var length = dataDefinition.PhysicalLength;
+                AppendToRow(start);
+                AppendToRow(GetEnd(start, length));
+                AppendToRow(length);
+
+                return row.ToString();
+
+                static bool IsIgnored(DataDefinition dataDefinition)
+                {
+                    DataDefinitionEntry codeElement = dataDefinition.CodeElement;
+                    return codeElement == null || codeElement.Line < 0;
+                }
+
+                void AppendToRow(object value)
+                {
+                    row.Append(value).Append(separator);
+                }
+
+                static string GetIndent(DataDefinition dataDefinition)
+                {
+                    var result = new StringBuilder();
+                    Node parentData = dataDefinition.Parent;
+                    while (parentData is not DataSection)
+                    {
+                        parentData = parentData.Parent;
+                        result.Append(INDENT);
+                    }
+
+                    return result.ToString();
+                }
+
+                static string GetName(DataDefinition dataDefinition)
+                {
+                    return dataDefinition.Name == null ? FILLER : dataDefinition.Name;
+                }
+
+                static string GetDeclaration(DataDefinition dataDefinition)
+                {
+                    string result = String.Empty;
+
+                    if (dataDefinition is DataRedefines dataRedefines)
+                    {
+                        result = string.Format(REDEFINES, dataRedefines.CodeElement.RedefinesDataName.Name);
+                    }
+                    else if (dataDefinition is DataDescription dataDescription && dataDescription.CodeElement is DataDescriptionEntry dataDescriptionEntry)
+                    {
+                        if (dataDescription.Picture != null)
+                        {
+                            result = string.Format(PIC, dataDescription.Picture.Value);
+                        }
+
+                        string usage = dataDescriptionEntry.Usage?.Token.Text;
+                        if (!String.IsNullOrWhiteSpace(usage))
+                        {
+                            AppendSpaceIfNotEmpty();
+                            result += usage;
+                        }
+
+                        if (result.Length == 0 && dataDefinition.ChildrenCount > 0)
+                        {
+                            result = GROUP;
+                        }
```

Comment:
Ok a group may not have any declaration but is it required to append one ?
The word 'GROUP' does not belong to the declarations in Cobol  so returning it may be confusing I don't know ?

Reply:
@fm-117  `GROUP` is required by the specifications ;)